Refactore/general improvments

talorlangnas-moveo opened the pull request.

talorlangnas-moveo committed:
feat: integrate axios for API calls and refactor chefs and restaurants pages

- Added axios as a dependency for handling HTTP requests.
- Refactored the ChefsPage to fetch chef data from an API and convert it to the required format.
- Updated the RestaurantsPage to fetch restaurant data from an API and include chef names in the display.
- Enhanced the TypeScript configuration to support new service paths.
- Removed hardcoded chef and restaurant data in favor of dynamic API calls.

talorlangnas-moveo committed:
refactor: update dish data handling and column definitions

- Renamed and updated the import paths for DishColumn and utility functions to reflect new structure.
- Enhanced the DishesPage to fetch and display dishes with additional restaurant and chef information.
- Removed deprecated dishes utility file and adjusted the dish conversion logic to include restaurant and chef names.
- Updated the DataTable to utilize the new data format for improved display of dish information.

talorlangnas-moveo committed:
Merge branch 'feature/server-update-api' into feature/admin-edit-add-delete-requests

talorlangnas-moveo committed:
refactor: streamline Home and Dishes components with updated data handling

- Replaced hardcoded card data in Home component with dynamic counts from the new adminStats utility.
- Updated DishesPage to remove deprecated dish data fetching logic and utilize the fetchAll function for improved data retrieval.
- Removed the restaurantsFilterOptions file to clean up unused code and enhance maintainability.

talorlangnas-moveo committed:
working on delete

talorlangnas-moveo committed:
feat: add new dependencies and enhance restaurant columns functionality

- Added `next-themes` and `sonner` packages to package.json for improved theming and notification support.
- Refactored restaurant columns to accept `onDelete` and `onEdit` props for better action handling.
- Updated RestaurantsPage to utilize the new RestaurantsTable component for displaying restaurant data.

talorlangnas-moveo committed:
refactor: update restaurant columns and enhance API functionality

- Renamed the restaurant columns file for clarity and updated the import path in RestaurantsTable.
- Removed the old restaurant columns implementation to streamline the codebase.
- Added an updateRestaurant function in the restaurants API service to handle restaurant updates with error handling and notifications.
- Introduced new dependencies for form handling and validation to improve the overall functionality.

talorlangnas-moveo committed:
working on create restaurant

talorlangnas-moveo committed:
adding create, edit, delete functions to Restaurant table page

- Simplified default values handling for create and update modes.
- Enhanced input placeholders and descriptions for better user guidance.
- Cleaned up unused properties in the restaurant interface and DTOs.

talorlangnas-moveo committed:
adding create, edit, delete functions to dish table page

refactor: update dish and restaurant handling with new features

- Added new Radix UI components for select and slider functionality in package.json.
- Removed deprecated columns file for dishes and replaced it with a new DishesTable component for better data display.
- Updated DishesPage to utilize the new DishesTable and improved data fetching logic.
- Enhanced dish API service with create, update, and delete functions, including error handling and notifications.
- Refactored dish and restaurant types to streamline data handling and improve maintainability.

talorlangnas-moveo committed:
adding create, edit, delete functions to chef table page

refactor: remove deprecated columns file and update ChefsPage to use ChefsTable

- Deleted the old columns file for chefs to streamline the codebase.
- Updated ChefsPage to utilize the new ChefsTable component for improved data display.
- Enhanced data fetching logic for chefs to ensure better performance and maintainability.

talorlangnas-moveo committed:
fix delete functions in chef and restaurant

refactor: enhance error handling and optional fields in API services

- Updated fetchChefById and fetchRestaurantById functions to return null on error instead of throwing exceptions, improving error handling.
- Made restaurantId and chefId optional in their respective schemas and DTOs to allow for more flexible data handling.
- Enhanced getChefsNameById and getRestaurantName functions to return default messages when no chef or restaurant is found.
- Updated the remove functions in both services to unset references in related documents upon deletion, ensuring data integrity.

talorlangnas-moveo committed:
refactor: update restaurant API functions to handle asynchronous card conversion

- Modified getNewestRestaurantsAsCards, getMostPopularRestaurantsAsCards, and getOpenRestaurantsAsCards functions to use Promise.all for converting restaurant data to card format, ensuring proper handling of asynchronous operations.

talorlangnas-moveo committed:
adding confirmation for delete action

- Changed the onDelete prop in DataTableRowAction and RestaurantColumns to return a Promise for better asynchronous handling.
- Replaced the delete form with a DeleteCard component for improved user experience.
- Removed unnecessary console log statements from DishesPage and RestaurantsPage to clean up the codebase.

talorlangnas-moveo committed:
refactor: enhance chef and dish image handling and API responses

- Updated image utility functions to return placeholder images when no valid image URL is found for chefs and dishes.
- Modified chef and dish API functions to use Promise.all for converting data to card format, ensuring proper asynchronous handling.
- Made imgUrl field optional in CreateChefDto, CreateDishDto, CreateRestaurantDto, and their corresponding schemas to allow for more flexible data handling.

talorlangnas-moveo committed:
fix: update ChefInfo import path for consistency

talorlangnas-moveo committed:
adding select chef list in restaurant-form

talorlangnas-moveo committed:
tmp commit

talorlangnas-moveo committed:
restaurant is ready

talorlangnas-moveo committed:
work on dishes

talorlangnas-moveo committed:
refactor: streamline dish management and enhance UI components

- Removed unnecessary props from DishColumns and integrated delete functionality directly within DataTableRowAction.
- Updated DishesLayout to fetch and display dishes and restaurants more efficiently.
- Simplified DishesPage by removing direct data fetching and utilizing context for data management.
- Enhanced DishForm to support dish creation and updates with improved context handling.
- Refactored DishesTable to utilize context for data retrieval, improving maintainability and readability.

talorlangnas-moveo committed:
chef is ready

refactor: enhance chef management and streamline components

- Removed unnecessary props from ChefColumns and integrated delete functionality directly within DataTableRowAction.
- Updated ChefsPage to eliminate direct data fetching, utilizing context for improved data management.
- Refactored ChefForm to support chef creation and updates with enhanced context handling.
- Simplified ChefsTable to leverage context for data retrieval, improving maintainability and readability.

talorlangnas-moveo committed:
refactor: improve chef and restaurant management with enhanced error handling and optional fields

- Updated chef image handling to allow undefined values for imgUrl in ChefForm.
- Modified chef and restaurant validation schemas to provide clearer error messages for missing references.
- Enhanced asynchronous handling in the chefs API by using Promise.all for converting chef data.
- Improved error messages in validation pipes for better user feedback when chefs or restaurants are not found.

talorlangnas-moveo committed:
delete restaurantContext

talorlangnas-moveo committed:
Merge branch 'feature/admin-edit-add-delete-requests' into refactore/general-improvments

talorlangnas-moveo committed:
Merge branch 'development' into refactore/general-improvments

taluliel18 commented on epicure-server/src/modules/dishes/pipes/validate-restaurant.pipe.ts:
```diff
@@ -11,21 +12,39 @@ export class ValidateRestaurantPipe implements PipeTransform<CreateDishDto> {
-    if (!value.restaurantId || !Types.ObjectId.isValid(value.restaurantId)) {
-      throw new NotFoundException('Invalid restaurant ID');
-    }
+  // async transform(value: CreateDishDto | UpdateDishDto) {
+  //   if (!value.restaurantId || !Types.ObjectId.isValid(value.restaurantId)) {
+  //     throw new NotFoundException('Invalid restaurant ID');
+  //   }
+
+  //   const restaurantExists = await this.restaurantModel.findById(
+  //     value.restaurantId,
+  //   );
+
+  //   if (!restaurantExists) {
+  //     throw new NotFoundException(
+  //       `Restaurant with ID ${value.restaurantId} not found`,
+  //     );
+  //   }
```
remove

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/services/chefs/image.utils.ts:
```diff
@@ -8,6 +8,7 @@ import OmerMillerImg from '@public/chefs/omer_miller.png';
 import ShahafShabatyImg from '@public/chefs/shahaf_shabaty.png';
 import NitzanRazImg from '@public/chefs/nitzan_raz.png';
 import YuvalBenNeriahImg from '@public/chefs/yuval_ben_neriah.png';
+// import chefPlaceholderImage from '@public/chefPlaceholder.png';
```
why you don’t use it?

talorlangnas-moveo replied:
I will use the placeholders images in the next branch

talorlangnas-moveo committed:
refactor: clean up unused code and improve restaurant validation logic

- Removed commented-out code for chef placeholder image in image.utils.ts.
- Cleaned up the ValidateRestaurantPipe by removing commented-out validation logic for restaurant ID checks, streamlining the transformation process for CreateDishDto and UpdateDishDto.
